src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -127,4 +135,37 @@
     return succeeded(records);
   }
 
+  Result<CheckInProcessRecords> sendItemStatusPatronNotice(CheckInProcessRecords records) {
+    RequestQueue requestQueue = records.getRequestQueue();
+    if (Objects.isNull(requestQueue)) {
+      return succeeded(records);
+    }
+
+    requestQueue.getRequests().stream()
+      .findFirst()
+      .ifPresent(
+        firstRequest -> {
+          String barcode = firstRequest.getRequesterBarcode();
+          userRepository.getUserByBarcode(barcode)
+            .thenAccept(r -> r.next(sendPatronNotice(records.getItem())));
+        }
+      );
+    return succeeded(records);
+  }
+
+  private Function<User, Result<User>> sendPatronNotice(Item item) {
+    return user -> {
+      if (AWAITING_PICKUP == item.getStatus()) {
```

Comment:
Please add method _isAwaitingPickup_ to Item and use it here

Reply:
done